src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +10,24 @@ public class ReadWriteList<E> {
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        lock.readLock().lock();
+        try {
+            list.add(element);
+        } finally {
+            lock.readLock().unlock();
+        }
```

Comment:
from `ReadLock.lock()` javadoc:
* If the **write** lock is held by another thread then the current thread becomes disabled for thread scheduling purposes and lies dormant until the read lock has been acquired.

which means that it is meant for use in the **read** operations, and `WriteLock.lock()` should be used in add/update/delete operations accordingly. Also a piece of `WriteLock.lock()` javadock:
* Acquires the write lock if neither the read nor write lock are held by another thread and returns immediately, setting the write lock hold count to one.